test_conformance/images/clFillImage/test_fill_generic.cpp

```diff
@@ -194,12 +194,26 @@ cl_mem create_image( cl_context context, cl_command_queue queue, BufferOwningPtr
     }
     else {
         // Else copy one scan line at a time.
+        size_t dstPitch2D = 0;
+        switch (imageInfo->type)
+        {
+            case CL_MEM_OBJECT_IMAGE3D:
+            case CL_MEM_OBJECT_IMAGE2D_ARRAY:
+            case CL_MEM_OBJECT_IMAGE2D:
+                dstPitch2D = mappedRow;
+                break;
+            case CL_MEM_OBJECT_IMAGE1D_ARRAY:
+            case CL_MEM_OBJECT_IMAGE1D:
```

Comment:
Why add IMAGE1D here? It just distracts from the real fix.

Reply:
dstPitch2D is initialised to 0 so need to set it to something valid for all cases in the switch.

There are other ways we could fix this bug, e.g. by swapping height and depth at line 148:

        case CL_MEM_OBJECT_IMAGE1D_ARRAY:
            height = imageInfo->arraySize;
            depth = 1;

(This was my original fix when I was fixing this for 1.2 but the fix proposed in the pull request matches the equivalent fix that is already present in clCopyImage/test_copy_generic.cpp)

Reply:
OK, sounds reasonable.

Reply:
Agree that this isn't right, that it should work and that reducing the diff with similar code is probably more helpful given the situation we're in.